server-web/.package-make-targets

```diff
@@ -0,0 +1,130 @@
+YELLOW := $(shell echo -e `tput setaf 3`)
+CYAN   := $(shell echo -e `tput setaf 6`)
+NC     := $(shell echo -e `tput setaf 9`)
+BOLD   := $(shell echo -e `tput bold`)
+SMUL   := $(shell echo -e `tput smul`)
+SGR0   := $(shell echo -e `tput sgr0`)
+
+export
+
+# function with a generic template to run docker with the required values
+# Accepts $1 = command to run, $2 = additional flags (optional)
+docker_run = docker run \
+    --name=$(DOCKER_NAME)-$@ \
+    --rm \
+    --user `id -u`:`id -g` \
+	--workdir=/usr/src/app/$(PACKAGE_NAME) \
+    -e CARGO_INCREMENTAL=$(CARGO_INCREMENTAL) \
+    -e RUSTC_BOOTSTRAP=$(RUSTC_BOOTSTRAP) \
+    -v "$(PWD)/../:/usr/src/app" \
+    -v "$(PWD)/../.cargo/registry:/usr/local/cargo/registry" \
+    $(2) \
+    -t $(BUILD_IMAGE_NAME):$(BUILD_IMAGE_TAG) \
+    $(1)
+
+ifeq ("$(CARGO_MANIFEST_PATH)", "")
+cargo_run = echo "$(BOLD)$(YELLOW)No Cargo.toml found in any of the subdirectories, skipping cargo check...$(NC)$(SGR0)"
+else
+cargo_run = $(call docker_run,cargo $(1) --manifest-path "$(CARGO_MANIFEST_PATH)" $(2))
+endif
+
+help:
+	@echo ""
+	@echo "$(BOLD)$(CYAN)Available targets$(NC)$(SGR0)"
+	@echo ""
+	@echo "$(SMUL)$(BOLD)$(GREEN)Rust$(NC)$(SGR0)"
+	@echo "  $(YELLOW)All cargo commands will use '--manifest-path $(CARGO_MANIFEST_PATH)'$(NC)"
+	@echo "  $(BOLD)build$(SGR0)       -- Run 'cargo build'"
+	@echo "  $(BOLD)release$(SGR0)     -- Run 'cargo build --release --target RELEASE_TARGET'"
+	@echo "                 (RELEASE_TARGET=$(RELEASE_TARGET))"
+	@echo "  $(BOLD)clean$(SGR0)       -- Run 'cargo clean'"
+	@echo "  $(BOLD)rust-check$(SGR0)  -- Run 'cargo check'"
+	@echo "  $(BOLD)rust-test$(SGR0)   -- Run 'cargo test --all'"
+	@echo "  $(BOLD)rust-clippy$(SGR0) -- Run 'cargo clippy --all -- -D warnings'"
+	@echo "  $(BOLD)rust-fmt$(SGR0)    -- Run 'cargo fmt --all -- --check' to check rust file formats."
+	@echo "  $(BOLD)rust-tidy$(SGR0)   -- Run 'cargo fmt --all' to fix rust file formats if needed."
+	@echo ""
+	@echo "$(SMUL)$(BOLD)$(GREEN)TOML$(NC)$(SGR0)"
+	@echo "  $(BOLD)toml-test$(SGR0)   -- Run 'taplo format --check' to validate TOML file formats."
+	@echo "  $(BOLD)toml-tidy$(SGR0)   -- Run 'taplo format' to fix TOML file formats if needed."
+	@echo ""
+	@echo "$(SMUL)$(BOLD)$(GREEN)Combined targets$(NC)$(SGR0)"
+	@echo "  $(BOLD)test$(SGR0) -- Run targets; rust-check rust-test rust-clippy rust-fmt toml-test"
+	@echo "  $(BOLD)tidy$(SGR0) -- Run targets; rust-tidy toml-tidy"
+
+	@echo "  $(BOLD)run$(SGR0)  -- Run docker container as a daemon, binding port $(HOST_PORT):$(DOCKER_PORT)"
+	@echo "  $(BOLD)stop$(SGR0) -- Run 'docker kill ${DOCKER_NAME}-run && docker rm ${DOCKER_NAME}-run' to stop and cleanup our docker after running"
+	@echo "  $(BOLD)all$(SGR0)  -- Run targets; test build release run"
+
+# Rust / cargo targets
+check-cargo-registry:
+	if [ ! -d "$(PWD)/../.cargo/registry" ]; then mkdir -p "$(PWD)/../.cargo/registry" ; fi
+
+build: check-cargo-registry
```

Comment:
do these need a docker-pull as well?

Reply:
Yeah, you're right. I should've taken some more time to fix the make file includes.
I've split them up a bit more now, so hopefully, it's a bit more readable and it's easier to make changes for all directories at once.